test/library/draft/Vector/Vector.chpl

```diff
@@ -0,0 +1,1264 @@
+
+      :arg other: The list to initialize from.
+    */
+    proc init=(other: list(this.type.eltType)) {
+      _checkType(this.type.eltType);
+      if !isCopyableType(this.type.eltType) then
+        compilerError("Cannot copy vector with element type that " +
+                      "cannot be copied");
+
+      this.eltType = this.type.eltType;
+      this.parSafe = this.type.parSafe;
+      this.complete();
+
+      _requestCapacity(other.size);
+      _commonInitFromIterable(other);
+    }
+
+    /*
+      Initializes a vector containing elements that are copied from
+      the elements contained in an array.
+
+      :arg other: The array to initialize from.
+    */
+    proc init=(other: [?d] this.type.eltType) {
+      _checkType(this.type.eltType);
+      if !isCopyableType(this.type.eltType) then
+        compilerError("Cannot copy vector with element type that " +
+                      "cannot be copied");
+      this.eltType = this.type.eltType;
+      this.parSafe = this.type.parSafe;
+      this.complete();
+
+      _requestCapacity(d.size);
+      _commonInitFromIterable(other);
+    }
+
+    /*
+      Initializes a vector containing elements that are copied from
+      the elements yielded by a range.
+
+      .. note::
+
+        Attempting to initialize a vector from an unbounded range will trigger
+        a compiler error.
+
+      :arg other: The range to initialize from.
+      :type other: `range(this.type.eltType)`
+    */
+    proc init=(other: range(this.type.eltType, ?b, ?d)) {
+      _checkType(this.type.eltType);
+      this.eltType = this.type.eltType;
+      this.parSafe = this.type.parSafe;
+
+      if !isBoundedRange(other) {
+        param e = this.type:string;
+        param f = other.type:string;
+        param msg = "Cannot init " + e + " from unbounded " + f;
+        compilerError(msg);
+      }
+
+      this.complete();
+
+      _requestCapacity(other.size);
+      _commonInitFromIterable(other);
+    }
+
+    pragma "no doc"
+    inline proc _enter() {
+      if parSafe then
+        _lock$.lock();
+    }
+
+    pragma "no doc"
+    inline proc _leave() {
+      if parSafe then
+        _lock$.unlock();
+    }
+
+    pragma "no doc"
+    proc _commonInitFromIterable(iterable) {
+      for x in iterable do {
+        _append(x);
+      }
+    }
+
+    pragma "no doc"
+    proc ref _append(x: eltType) lifetime x > this {
+      _requestCapacity(_size+1);
+      _data[_size] = x;
+      _size += 1;
+    }
+
+    pragma "no doc"
+    proc ref _append(ref x: eltType) where isOwnedClass(x)
+    lifetime x > this {
+      _requestCapacity(_size+1);
+      _data[_size] = x;
+      _size += 1;
+    }
+
+    /*
+      The current number of elements contained in this vector.
+    */
+    inline proc const size {
+      var result = 0;
+
+      on this {
+        _enter();
+        result = _size;
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      The capacity of this vector.
```

Comment:
Were you going to elaborate on "capacity"?

Reply:
https://github.com/chapel-lang/chapel/pull/16048/commits/a7c94bbbe715ff5271aebb8fad64b0bcabe83121

Reply:
Too many lines and changes...